test/unit/simulator-management-specs.js

```diff
@@ -86,4 +86,80 @@ describe('simulator management', function () {
       getSimulatorStub.notCalled.should.be.true;
     });
   });
+  describe('runSimulatorReset', function () {
+    let result;
+    const stoppedDeviceDummy = {
+      isRunning: () => false,
+      scrubCustomApp: (path, bundleId) => {
+        result = {path, bundleId};
+      },
+      clean: () => {
+        result = 'cleaned';
+      },
+      shutdown: () => {}
+    };
+
+    beforeEach(function () {
+      result = undefined;
+    });
+
+    it('should call scrubCustomApp with fastReset', async function () {
+      const opts = {
+        udid: '301CD634-00A9-4042-B463-BD4E755167EA',
+        bundleId: 'io.appium.example',
```

Comment:
One think you might want to clarify in these tests, is the fact that executions do not require BOTH app+bundleId , but rather  need either/or .   I notice all these tests specify both, which is not necessarily the same as users will be executing with.

Reply:
if bundleId is not provided explicitly then Appium will anyway try to parse it from the package info. Setting it explicitly allows to save some time (usually less than a second) for session init

Reply:
Yes. bundleId was parsed in a part of the internal process from `app`.